src/hdmf/spec/namespace.py

```diff
@@ -18,7 +18,7 @@
     {'name': 'name', 'type': str, 'doc': 'the name of this namespace'},
     {'name': 'schema', 'type': list, 'doc': 'location of schema specification files or other Namespaces'},
     {'name': 'full_name', 'type': str, 'doc': 'extended full name of this namespace', 'default': None},
-    {'name': 'version', 'type': (str, tuple, list), 'doc': 'Version number of the namespace', 'default': None},
+    {'name': 'version', 'type': (str, tuple, list), 'doc': 'Version number of the namespace'},
```

Comment:
Because `'full_name'` is after an optional argument, this change will invisibly break code that uses only positional arguments. For example:
```python
ns = SpecNamespace('doc', 'name', [{'source': 'test.yaml'}], 'full name', 'version')
ns
```
old output:
```
{'doc': 'doc', 'schema': [{'source': 'test.yaml'}], 'name': 'name', 'full_name': 'full name', 'version': 'version'}
```
new output:
```
{'doc': 'doc', 'schema': [{'source': 'test.yaml'}], 'name': 'name', 'full_name': 'version', 'version': 'full name'}
```

So I am more in favor of keeping `'default': None` and checking for None in the constructor and raising an error there if it is none.

Reply:
But since this is internal code that I believe is not used by anything outside of PyNWB/HDMF, I think this is fine. I would just reorder the docval args to make it clear that this is the new order of positional arguments.